internal/proxyservice/tcp/pg/backend.go

```diff
@@ -2,51 +2,46 @@ package pg
 
 import (
 	"fmt"
-	"log"
 	"net"
 	"reflect"
 
-	"github.com/cyberark/secretless-broker/internal/handlers/pg/protocol"
 	"github.com/cyberark/secretless-broker/internal/handlers/ssl"
-	"github.com/cyberark/secretless-broker/internal/util"
+	"github.com/cyberark/secretless-broker/internal/proxyservice/tcp/pg/protocol"
 )
 
 // ConfigureBackend resolves the backend connection settings and credentials and sets the
 // connectionDetails field.
-func (h *Handler) ConfigureBackend() error {
+func (s *SingleUseConnector) ConfigureBackend(credentials map[string][]byte) error {
 	var err error
-	var options map[string][]byte
 
-	if options, err = h.Resolver.Resolve(h.GetConfig().Credentials); err != nil {
-		return err
-	}
+	keys := reflect.ValueOf(credentials).MapKeys()
+	s.logger.Debugf("backend connection parameters: %s", keys)
 
-	if h.GetConfig().Debug {
-		keys := reflect.ValueOf(options).MapKeys()
-		log.Printf("%s backend connection parameters: %s", h.GetConfig().Name, keys)
+	if len(credentials["address"]) > 0 {
+		s.logger.Warnf("'address' has been deprecated for PG connector. " +
+			"Please use 'host' and 'port' instead.'")
 	}
 
-	backendConfig, err := NewBackendConfig(options)
+	backendConfig, err := NewBackendConfig(credentials)
 	if err != nil {
 		return err
 	}
 
-	h.BackendConfig = backendConfig
+	s.backendConfig = backendConfig
 
 	return nil
 }
 
 // ConnectToBackend establishes the connection to the backend database and sets the Backend field.
-func (h *Handler) ConnectToBackend() (err error) {
+func (s *SingleUseConnector) ConnectToBackend() (err error) {
 	var connection net.Conn
-	if connection, err = net.Dial("tcp", h.BackendConfig.Address()); err != nil {
+	if connection, err = net.Dial("tcp", s.backendConfig.Address()); err != nil {
 		return
 	}
 
-	debug := util.OptionalDebug(h.GetConfig().Debug)
-	debug("Sending startup message")
+	s.logger.Debugln("Sending startup message")
 
-	tlsConf, err := ssl.NewDbSSLMode(h.BackendConfig.SSLOptions, true)
+	tlsConf, err := ssl.NewDbSSLMode(s.backendConfig.SSLOptions, true)
```

Comment:
Old code I know, and not a blocker, but if there's a clean way to pull out that `if tlsConf.UseTLS {` block below (I didn't give it too much thought) it would be nice to.